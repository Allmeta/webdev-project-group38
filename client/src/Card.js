import React, { Component } from 'react'
import styled from 'styled-components'

class Card extends Component{
    render(){

    }
}
export default Card