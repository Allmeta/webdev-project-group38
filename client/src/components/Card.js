import React, { Component } from 'react'
import { Label } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import Comment from './CardComment'
import styled, { css } from 'styled-components'

const Card = styled.div`
  display:inline-grid;
  grid-template-columns: 185px auto;
  color:black;
  box-shadow: 0 2px 20px rgba(49,54,68,.3);
  background:#fafafa;
  min-height:265px;
  min-width:370px;
  box-sizing:border-box;
`
const CardImg = styled.div`
  background-position: 50%;
  background-repeat: no-repeat;
  background-size: cover;
  position:relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-image: url(${props => props.img});
`
const CardTitle = styled.div`
  background:rgba(0,0,0,.5);
  color:white;
  width:100%;
  padding:12px;
  text-align:center;
  font-size:14px;
`
const CardData = styled.div`
  position:relative;
`
const CardDate = styled.div`
  text-align:center;
  font-size:14px;
  padding:10px;
  background:#f1f3f7;
  color:#4553F4;
  font-weight:bold;
`
const CardExtra = styled.div`
  color:rgb(122,133,143);
  display: grid;
  font-size: 13px;
  grid-template-columns: repeat(2,1fr);
  background:#f6f8fb;
`
const CardSpan = styled.span`
  padding:8px;
  text-align:center;
  display:inline-block;
`
const CardDesc = styled.div`
  color:rgb(122,133,143);
  height:160px;
  background:#fafafa;
  line-height:15px;
  padding:4px 10px;
  font-size:12px;
  ${props => props.show === false
    ? css`
    overflow-y:auto;
  ` : css`
    overflow-y:hidden;
  `}
`
const CardBottom = styled.div`
  color: rgb(146,153,161);
  line-height: 13px;
  padding: 8px 10px;
  max-height:42px;
  text-align: center;
  width: 100%;
  background:#f6f8fb;
  ${props => !props.show && css`
  display:none;
  `}
`

class MovieCard extends Component {
  constructor (props) {
    super(props)
    this.state = {
      toggled: true
    }
    this.mouseIn = this.mouseIn.bind(this)
    this.mouseOut = this.mouseOut.bind(this)
  }
  mouseIn () {
    this.setState({
      toggled: false
    })
  }
  mouseOut () {
    this.setState({
      toggled: true
    })
  }

  render () {
    return (
      <Card onMouseEnter={this.mouseIn} onMouseLeave={this.mouseOut}>
        <CardImg img={this.props.img}>
          <CardTitle>{this.props.title}</CardTitle>
        </CardImg>
        <CardData>
          <CardDate>{this.props.date}</CardDate>
          <CardExtra>
            <CardSpan><Label content={this.props.language} icon="location arrow"/></CardSpan>
            <CardSpan><Label content={this.props.popularity} icon="star"></Label></CardSpan>
          </CardExtra>
          <CardDesc show={this.state.toggled}>{this.props.description}</CardDesc>
          <CardBottom show={this.state.toggled}>
            {this.props.genres.map((o) => <Label key={'' + o} content={o}/>)}
          </CardBottom>
          <CardBottom show={!this.state.toggled}>
            <Comment comment={this.props.comment}></Comment>
          </CardBottom>
        </CardData>
      </Card>

    )
  }
}

MovieCard.propTypes = {
  popularity: PropTypes.number.isRequired,
  description: PropTypes.string,
  img: PropTypes.string,
  title: PropTypes.string.isRequired,
  genres: PropTypes.array.isRequired,
  comment: PropTypes.string,
  date: PropTypes.string,
  language: PropTypes.string
}

export default MovieCard
