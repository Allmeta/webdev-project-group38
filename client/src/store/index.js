import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;