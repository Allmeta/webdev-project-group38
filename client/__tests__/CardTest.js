import React from 'react'
import Card from '../src/Components/Card.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../src/store/index.js'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Card title="snapshot" popularity={123} genres={['snapshot genre']} /></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
