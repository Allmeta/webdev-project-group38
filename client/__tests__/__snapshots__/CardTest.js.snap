// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui blue card"
  onClick={[Function]}
>
  <img
    className="ui image"
    onClick={[Function]}
  />
  <div
    className="content"
    onClick={[Function]}
  >
    <div
      className="header"
    >
      snapshot
    </div>
    <div
      className="meta"
    >
      <span
        className="date"
      >
          
        snapshot genre
         
      </span>
    </div>
  </div>
  <div>
    <div
      className="ui blue icon fade animated bottom attached button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="visible content"
      >
        <i
          aria-hidden="true"
          className="blue add circular icon"
          style={
            Object {
              "background": "white",
            }
          }
        />
      </div>
      <div
        className="hidden content"
      >
        Comment on movie
      </div>
    </div>
  </div>
  <div
    className="extra content"
  >
    <span>
      <i
        aria-hidden="true"
        className="star icon"
      />
       
      123
    </span>
  </div>
</div>
`;
